### PR TITLE
Juanro/new hire bug fix

### PR DESCRIPTION
**Pull request by Juanro-Minne**
Added a check for a case where the employee address id is 0  fixing the null reference error.

### CODE REVIEW

**Comment by MTSchoeman** [comment]
Unit Test Coverage report

**Comment by Juanro-Minne** [comment]
Screenshot 2024-02-22 111632

**Comment by Juanro-Minne** [comment]
> Unit Test Coverage report

Added

**Review by ReinoRabbit — DISMISSED** [review]

**Review by MTSchoeman — APPROVED** [review]

File: RGO.Services.Tests/Services/EmployeeAddressServiceUnitTest.cs [hunk 1]
```diff
@@ -59,6 +59,18 @@ public async Task CheckIfExistsPassTest()
         Assert.True(result);
     }
 
+    [Fact]
+    public async Task CheckIfExistPassFail()
```
Comment (AWallsRR) [inline RGO.Services.Tests/Services/EmployeeAddressServiceUnitTest.cs:63]: Just one thing does the name of this function make sense?

Comment (Juanro-Minne) [reply RGO.Services.Tests/Services/EmployeeAddressServiceUnitTest.cs:63]: No really ill change it to this (CheckIfExistsFail)

**Review by AWallsRR — APPROVED** [review]

**Review by ReinoRabbit — APPROVED** [review]

**Review by MTSchoeman — APPROVED** [review]

File: RGO.Services/Services/EmployeeAddressService.cs
[the remark this answers is not in the document]
Comment (dvanhout89) [reply RGO.Services/Services/EmployeeAddressService.cs:1]: @Juanro-Minne

File: RGO.Services/Services/EmployeeAddressService.cs [hunk 2]
```diff
@@ -16,6 +16,9 @@ public EmployeeAddressService(IUnitOfWork db)
 
     public async Task<bool> CheckIfExists(EmployeeAddressDto employeeAddressDto)
     {
+        if(employeeAddressDto.Id == 0) {
+            return false;
+        }
         var exists = await _db.EmployeeAddress.GetById(employeeAddressDto.Id);
         return exists != null;
```
[the remark this answers is not in the document]
Comment (dvanhout89) [reply RGO.Services/Services/EmployeeAddressService.cs:23]: @Juanro-Minne

**Review by dvanhout89 — APPROVED** [review]

**Pull request merged**